# Ejercicio 1

Se pretende estimar los valores de producción Y (en miles de toneladas) de cierto material, en función del tiempo
transcurrido X (en meses) usando los valores de la tabla:
| X  |  Y  |
|----|-----|
|  1 |   9 |
|  8 |  25 |
| 10 | 180 |
| 16 | 326 |
| 22 | 538 |


Se plantea un modelo de la forma Y = a + bx + cx2
. Encontrar los estimadores de mínimos cuadrados para a, b y c en este
modelo

In [6]:
#Código Python de resolución de ejercicio 1

import numpy as np

#Datos

X = np.array([1,8,10,16,22])
y = np.array([9,25,180,326,538])


#Matríz de diseño

X_matrix = np.vstack([np.ones(len(X)), X, X**2]).T


#Solución aplicando mínimos cuadrados

params = np.linalg.lstsq(X_matrix, y, rcond=None)[0]

a, b, c = params

print(f'a: {a} b:{b} c:{c}')



a: -10.4448001959549 b:5.43606970396827 c:0.9064840086779943


# Estimación de los parámetros del modelo \(Y = a + bX + cX^2\)

## Datos
Los datos proporcionados son:
$$
\begin{aligned}
X & : \{1, 8, 10, 16, 22\} \\
Y & : \{9, 25, 180, 326, 538\}
\end{aligned}
$$

## Modelo
El modelo planteado es:
$$
Y = a + bX + cX^2
$$

## Formulación matricial
Para encontrar los estimadores de mínimos cuadrados, formulamos el problema en términos matriciales. El modelo se puede escribir como:
$$
\mathbf{Y} = \mathbf{X} \beta + \epsilon
$$
donde:
$$
\mathbf{X} = \begin{pmatrix}
1 & 1 & 1^2 \\
1 & 8 & 8^2 \\
1 & 10 & 10^2 \\
1 & 16 & 16^2 \\
1 & 22 & 22^2
\end{pmatrix} = \begin{pmatrix}
1 & 1 & 1 \\
1 & 8 & 64 \\
1 & 10 & 100 \\
1 & 16 & 256 \\
1 & 22 & 484
\end{pmatrix}, \quad 
\beta = \begin{pmatrix}
a \\
b \\
c
\end{pmatrix}, \quad 
\mathbf{Y} = \begin{pmatrix}
9 \\
25 \\
180 \\
326 \\
538
\end{pmatrix}
$$

## Cálculo de los estimadores de mínimos cuadrados
Los estimadores se obtienen resolviendo el sistema de ecuaciones normales:
$$
(\mathbf{X}^T \mathbf{X}) \beta = \mathbf{X}^T \mathbf{Y}
$$

Primero calculamos $ (\mathbf{X}^T \mathbf{X}) y (\mathbf{X}^T \mathbf{Y})$

$$
\mathbf{X}^T \mathbf{X} = \begin{pmatrix}
1 & 1 & 1 & 1 & 1 \\
1 & 8 & 10 & 16 & 22 \\
1 & 64 & 100 & 256 & 484
\end{pmatrix}
\begin{pmatrix}
1 & 1 & 1^2 \\
1 & 8 & 8^2 \\
1 & 10 & 10^2 \\
1 & 16 & 16^2 \\
1 & 22 & 22^2
\end{pmatrix} = \begin{pmatrix}
5 & 57 & 905 \\
57 & 905 & 16211 \\
905 & 16211 & 305335
\end{pmatrix}
$$

$$
\mathbf{X}^T \mathbf{Y} = \begin{pmatrix}
1 & 1 & 1 & 1 & 1 \\
1 & 8 & 10 & 16 & 22 \\
1 & 64 & 100 & 256 & 484
\end{pmatrix}
\begin{pmatrix}
9 \\
25 \\
180 \\
326 \\
538
\end{pmatrix} = \begin{pmatrix}
1078 \\
15777 \\
300653
\end{pmatrix}
$$

Luego, resolvemos para $\beta$:
$$
\beta = (\mathbf{X}^T \mathbf{X})^{-1} \mathbf{X}^T \mathbf{Y}
$$

Calculamos la inversa de $(\mathbf{X}^T \mathbf{X})$:
$$
(\mathbf{X}^T \mathbf{X})^{-1} = \begin{pmatrix}
5 & 57 & 905 \\
57 & 905 & 16211 \\
905 & 16211 & 305335
\end{pmatrix}^{-1}
$$

Finalmente, multiplicamos:
$$
\beta = \begin{pmatrix}
-10.4448 \\
5.4361 \\
0.9065
\end{pmatrix}
$$

## Resultados
Los estimadores de mínimos cuadrados para los parámetros (a\), (b\) y (c\) son:
$$
\begin{aligned}
a &= -10.4448 \\
b &= 5.4361 \\
c &= 0.9065 \\
\end{aligned}
$$

Por lo tanto, el modelo ajustado es:
$$
Y = -10.4448 + 5.4361X + 0.9065X^2
$$



# Ejercicio 2

Una empresa aseguradora desea estimar el porcentaje de morosidad en una determinada cartera de gran dimensión que tiene emitidas 8 pólizas, de las cuales 2 están en mora. El investigador encargado, teniendo en cuenta toda la información disponible y su conocimiento sobre el funcionamiento interno de la empresa, considera que puede modelar el porcentaje p de morosidad, según una distribución $\Beta$ (1, 2). Determinar la distribución a posteriori del parámetro p.
Determinar su media y su varianza.

In [18]:
from scipy.stats import beta

#Parámetros de la distribución beta posterior
alpha_post =1+2
beta_post = 2+8-2

#Media y varianza de la distribución beta
mean_post = alpha_post / (alpha_post+beta_post)
variance_post = (alpha_post*beta_post)/(((alpha_post+beta_post)**2)*(alpha_post+beta_post+1))

print(f'Media: {mean_post}')
print(f'Varianza: {variance_post}')



 

Media: 0.2727272727272727
Varianza: 0.01652892561983471


# Estimación del Porcentaje de Morosidad

## Datos del Problema
Una empresa aseguradora desea estimar el porcentaje de morosidad en una cartera de gran dimensión que tiene emitidas 8 pólizas, de las cuales 2 están en mora. El investigador considera que puede modelar el porcentaje \(p\) de morosidad según una distribución \(B(1, 2)\).

## Modelo a Priori
El investigador modela el porcentaje \(p\) de morosidad según una distribución Beta. La distribución Beta se define como:
$$
p \sim \text{Beta}(\alpha, \beta)
$$
donde los parámetros son:
$$
\alpha = 1
$$
$$
\beta = 2
$$

## Datos Observados
Los datos observados de la cartera de pólizas son:
- Número total de pólizas: \(n = 8\)
- Número de pólizas en mora: \(x = 2\)

## Distribución a Posteriori
Para obtener la distribución a posteriori del parámetro \(p\) dado los datos observados, utilizamos la fórmula de Bayes. La fórmula de Bayes para actualizar la distribución Beta a priori con datos observados sigue resultando en una distribución Beta.

### Paso 1: Definir la Distribución a Posteriori
Dado que la distribución a priori es una distribución Beta y los datos observados siguen una distribución binomial, la distribución a posteriori también será una distribución Beta. La forma general es:
$$
p | x \sim \text{Beta}(\alpha + x, \beta + n - x)
$$

### Paso 2: Calcular los Nuevos Parámetros
Usamos los datos observados y los parámetros de la distribución a priori para calcular los nuevos parámetros \(\alpha_{\text{posteriori}}\) y \(\beta_{\text{posteriori}}\):

1. Actualizar \(\alpha\) con los éxitos observados \(x\):
$$
\alpha_{\text{posteriori}} = \alpha + x = 1 + 2 = 3
$$

2. Actualizar \(\beta\) con los fracasos observados \(n - x\):
$$
\beta_{\text{posteriori}} = \beta + n - x = 2 + 8 - 2 = 8
$$

### Paso 3: Escribir la Distribución a Posteriori
Por lo tanto, la distribución a posteriori es:
$$
p | x \sim \text{Beta}(3, 8)
$$

## Media y Varianza de la Distribución a Posteriori
La media y la varianza de una distribución Beta(\(\alpha\), \(\beta\)) se calculan usando las siguientes fórmulas:

### Paso 4: Calcular la Media
La media de una distribución Beta es:
$$
\text{Media} = \frac{\alpha}{\alpha + \beta}
$$
Sustituyendo los valores de \(\alpha_{\text{posteriori}}\) y \(\beta_{\text{posteriori}}\):
$$
\text{Media} = \frac{3}{3 + 8} = \frac{3}{11}
$$

### Paso 5: Calcular la Varianza
La varianza de una distribución Beta es:
$$
\text{Varianza} = \frac{\alpha \beta}{(\alpha + \beta)^2 (\alpha + \beta + 1)}
$$
Sustituyendo los valores de \(\alpha_{\text{posteriori}}\) y \(\beta_{\text{posteriori}}\):
$$
\text{Varianza} = \frac{3 \cdot 8}{(3 + 8)^2 (3 + 8 + 1)} = \frac{24}{11^2 \cdot 12} = \frac{24}{121 \cdot 12} = \frac{24}{1452} = \frac{1}{66}
$$

## Resultados
- La distribución a posteriori del parámetro \(p\) es $ \text{Beta}(3, 8) $.
- La media de la distribución a posteriori es $ \frac{3}{11} \approx 0.2727 $.
- La varianza de la distribución a posteriori es $ \frac{1}{66} \approx 0.0152 $.
